### PR TITLE
KAFKA-2809: Improve documentation linking

### PR DESCRIPTION
**Pull request by granthenke**
Often it is useful to link to a specific header within the documentation. Especially when referencing docs in the mailing lists.

This adds anchors and links for all headers in the docs.

### CODE REVIEW

**Comment by granthenke** [comment]
@junrao @gwenshap @guozhangwang  Would any of you have a chance to review this?

**Comment by junrao** [comment]
Thanks for the patch. Could you rebase?

**Comment by junrao** [comment]
@granthenke : Do you have time to work on this today so that it can be included in the 0.9.0.0 release? It seems that we may need to roll out RC3 due to KAFKA-2841.

**Comment by granthenke** [comment]
@junrao Rebased the patch. Thanks for looking at it.

File: docs/ops.html [hunk 1]
```diff
@@ -426,8 +426,8 @@
     <li>We upped the max socket buffer size to enable high-performance data transfer between data centers <a href="http://www.psc.edu/index.php/networking/641-tcp-tune">described here</a>.
 </ul>
 
-<h4><a id="diskandfs">Disks and Filesystem</a></h4>
-We recommend using multiple drives to get good throughput and not sharing the same drives used for Kafka data with application logs or other OS filesystem activity to ensure good latency. You can either RAID these drives together into a single volume or format and mount each drive as its own directory. Since Kafka has replication the redundancy provided by RAID can also be provided at the application level. This choice has several tradeoffs.
+<h4><a id="diskandfs" href="#diskandfs">Disks and Filesystem</a></h4>
```
Comment (junrao) [inline docs/ops.html:429]: 0.8 should be 0.9, right?

File: docs/ops.html [hunk 2]
```diff
@@ -129,13 +129,13 @@
...
 <p>
 The process of migrating data is manually initiated but fully automated. Under the covers what happens is that Kafka will add the new server as a follower of the partition it is migrating and allow it to fully replicate the existing data in that partition. When the new server has fully replicated the contents of this partition and joined the in-sync replica one of the existing replicas will delete their partition's data.
 <p>
-The partition reassignment tool can be used to move partitions across brokers. An ideal partition distribution would ensure even data load and partition sizes across all brokers.
+The partition reassignment tool can be used to move partitions across brokers. An ideal partition distribution would ensure even data load and partition sizes across all brokers. In 0.8.1, the partition reassignment tool does not have the capability to automatically study the data distribution in a Kafka cluster and move partitions around to attain an even load distribution. As such, the admin has to figure out which topics or partitions should be moved around.
```
Comment (junrao) [inline docs/ops.html]: Should we reference 0.9.0 instead of 0.8.1? Or we can just remove the version.

**Comment by junrao** [comment]
Some of the hrefs reference "#" and some other don't. Should we make them consistent?

**Comment by granthenke** [comment]
@junrao Updated the patch to fix the links to use # and remove the old version references.

**Comment by junrao** [comment]
Thanks for the patch. LGTM.

**Comment by sm4rk0** [comment]
@guozhangwang sorry for using this way of communication. I've just dropped by freenode chatroom and found it's pretty unused... Here's a copy-paste from there:
Hi guys,
I'm reading documentation here: https://kafka.apache.org/documentation
and I've noticed that clicking the chapters in menu on the left, the page always reload
I believe this is because links in navigation menu don't include trailing "/" that is added when the page is loaded
for example, "Kafka Connect" chapter has this url: https://kafka.apache.org/documentation#connect
but when the page loads, the address bar (in Firefox) contains this: https://kafka.apache.org/documentation/#connect

**Comment by guozhangwang** [comment]
cc @derrickdoo , maybe you can address this question?

**Pull request closed without merging**